tangelo/linq/target/target_qiskit.py

```diff
@@ -92,11 +96,34 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
             job_sim = self.qiskit.execute(translated_circuit, backend, noise_model=qiskit_noise_model,
                                           shots=self.n_shots, basis_gates=None, optimization_level=opt_level)
             sim_results = job_sim.result()
-            frequencies = {state[::-1]: count/self.n_shots for state, count in sim_results.get_counts(0).items()}
+            self.all_frequencies = {state[::-1]: count/self.n_shots for state, count in sim_results.get_counts(0).items()}
 
-            self.all_frequencies = frequencies.copy()
+            frequencies = self.all_frequencies
             self._current_state = None
 
+        # desired_meas_result is not None and return_statevector is requested so loop shot by shot (much slower)
```

Comment:
Discrepancy between L104 and 105 ? Does it even make sense to return a state representation unless its a density matrix, for a mixed state ?